pkg/actor/deploy_test.go

```diff
@@ -73,7 +73,7 @@ func TestDeploysNotInitalizedClusterAfterVersionChecker(t *testing.T) {
 	cluster.SetTrue(api.CrdbVersionChecked)
 
 	mock := kube.MockKubernetesDistribution()
-	deploy := actor.NewDeploy(scheme, client, nil, mock)
+	deploy := actor.NewDeploy(scheme, client, nil, mock, nil)
```

Comment:
rather than passing `nil` for unused values could we update the signatures? We require that these functions return `Actor` instances, but we don't need to satisfy any other interface right?

Reply:
I did a (relatively naive) audit of all the signatures of the actor constructors and cleaned up a lot. There are still some instances in the tests where pass a `nil` but the actor requires it, and we're just not testing some specific flow.